applications/vessel-history/src/features/vessels/activity/vessels-activity.thunk.ts

```diff
@@ -44,15 +44,26 @@ const groupEvents = (events: ActivityEvent[]) => {
         if (
             !groups.length ||
             groups[groups.length - 1].event_type !== event.type ||
-            groups[groups.length - 1].event_place !== event.regions.eez[0]
+            (
+                groups[groups.length - 1].event_eez !== event.regions.eez[0] ||
+                groups[groups.length - 1].event_rfmo !== event.regions.rfmo[0]
+            )
         ) {
             groups.push({
                 event_type: event.type,
-                event_place: event.regions.eez[0],
+                event_eez: event.regions.eez[0],
+                event_rfmo: event.regions.rfmo[0],
+                ocean: event.regions.ocean[0],
                 open: true,
                 entries: [event]
             })
-        } else if (groups[groups.length - 1].event_type === event.type && groups[groups.length - 1].event_place === event.regions.eez[0]) {
+        } else if (
+            groups[groups.length - 1].event_type === event.type &&
+            (
+                groups[groups.length - 1].event_eez === event.regions.eez[0] &&
+                groups[groups.length - 1].event_rfmo === event.regions.rfmo[0]
```

Comment:
What about the `ocean`? shouldn't it be verified too?

Reply:
because at the moment I did this, I asked for the criteria to create the grops but is not defined, so did ot based on what I can see in the design and the data, maybe this can change in the future or they like my solution